programs/whirlpool/src/instructions/v2/set_config_extension_authority.rs

```diff
@@ -0,0 +1,25 @@
+use anchor_lang::prelude::*;
+
+use crate::state::{WhirlpoolsConfig, WhirlpoolsConfigExtension};
+
+#[derive(Accounts)]
+pub struct SetConfigExtensionAuthority<'info> {
+    pub whirlpools_config: Box<Account<'info, WhirlpoolsConfig>>,
+
+    #[account(mut, has_one = whirlpools_config)]
+    pub whirlpools_config_extension: Account<'info, WhirlpoolsConfigExtension>,
+
+    #[account(address = whirlpools_config_extension.config_extension_authority)]
+    pub config_extension_authority: Signer<'info>,
+
+    /// CHECK: safe, the account that will be new authority can be arbitrary
+    pub new_config_extension_authority: UncheckedAccount<'info>,
+}
+
+/// Set the fee authority. Only the current config extension has permission to invoke this instruction.
+pub fn handler(ctx: Context<SetConfigExtensionAuthority>) -> Result<()> {
+    Ok(ctx
+        .accounts
+        .whirlpools_config_extension
+        .update_config_extension_authority(ctx.accounts.new_config_extension_authority.key()))
```

Comment:
I think we are talking about `set_reward_authority_by_super_authority` and `set_reward_emissions_super_authority`.

In my understanding, `reward_authority` for each reward is set on Whirlpool account.
I think super_authority is a mechanism for "delegation" and to bail out in case the "delegated" third party loses or misuses the authority.
So I don't think it is used to rescue the case where an important authority included in WhirlpoolsConfig is mistakenly set.

Now there is no instruction to rescue the case where the authority defined in WhirlpoolsConfig is mistakenly set.
Therefore, I think the same should be applied to the authority of WhirlpoolsConfigExtension.
(If such a situation should occur, I think we need to add a rescue instruction.)